diff -ruN benchmark/libming-4.7/util/decompile.c benchmark/libming-4.7-patched/libming-4.7-patch-2017-11728/util/decompile.c
--- benchmark/libming-4.7/util/decompile.c	2022-11-22 17:20:19.749006812 +0900
+++ benchmark/libming-4.7-patched/libming-4.7-patch-2017-11728/util/decompile.c	2022-11-05 15:03:05.972141865 +0900
@@ -864,7 +864,13 @@
 		SWF_warn("OpCode: want %i, max %i\n", n, maxn);
 #endif
 		return -999;
-	}
+	} else if (n < 1) {
+
+#if DEBUG
+		SWF_warn("OpCode: want %i < 1\n", n);
+#endif
+		return -998;
+        }
 	return actions[n].SWF_ACTIONRECORD.ActionCode;
 }
 
